tasks/rpmbuild-test

```diff
@@ -25,6 +25,24 @@ if [ "${fed_branch}" = "master" ]; then
 fi
 mkdir ${RSYNC_BRANCH}
 mkdir repo
+# Write uuid to a lock file and store a backup
+uuidgen > file.lock
+cp file.lock uuid.saved
+while false; do
+     # Check if lock exists on remote server
+     while [[ $(rsync --dry-run -avz file.lock fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir) != *"file.lock"* ]]; do
+          sleep 60
+     done
+     # Push lock file with uuid to remote server
+     rsync -avz file.lock fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir/
+     # Pull lock file back
+     rsync -avz fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir/file.lock file.lock
+     # If uuid matches, we can proceed
+     if [[ $(diff file.lock uuid.saved) == "" ]]; then
+          break
+     fi
+     sleep 60
+done
```

Comment:
You should define a trap function to remove the lock after getting it.  Do this instead of lines 60 and 61.  This way if anything breaks the script it will still remove the lock file.
```
# Kill backgrounded jobs on exit
function clean_up {
# Delete the rsync lock we placed
rsync -vr --delete $(mktemp -d)/ fedora-atomic@artifacts.ci.centos.org::fedora-atomic/${RSYNC_BRANCH}/repo/lockdir/
}
trap clean_up EXIT SIGHUP SIGINT SIGTERM

```